### ISSUE TITLE
license

### ISSUE THREAD

**Issue by aprive** [KlugerLab/FIt-SNE#24]
Very nice work! Have you decided how to license this software?

**Comment by linqiaozhi** [comment]
Thanks, @aprive! I am happy for it to be simple and permissive (e.g. MIT license). I did not officially add it because I wasn't sure how--much of the code is derived from the original BH t-SNE code, so it still has the old license on it. I was not sure how to add my own license then on top of that. 

Do you know how this works? I presume I should just keep the copyright notices in each file and add another on top of that?

**Comment by aprive** [comment]
Okay, great!

It looks like in addition to the original t-SNE code, the other licenses used in the software include GPL for FFTW and Apache 2.0 for annoylib. From what I understand, software under the Apache 2.0 license can be packaged together with software under the [GPL >= 3](https://en.wikipedia.org/wiki/Apache_License#GPL_compatibility), and the FFTW GPL 2 license [states](https://github.com/FFTW/fftw3/blob/master/COPYRIGHT) that the GPL version used can be 3. Also, if you go with the permissive MIT license, I think that can be packaged with GPL-3 code.

What is the license of the original BH t-SNE code (also MIT)? When I search "license" in this repository, I don't see any other than what I mentioned in the above paragraph.

To be honest, I can't claim to precisely understand all legal implications of mixing the permissive licenses MIT and Apache 2.0 with the copyleft GPL (e.g., what is said [here](https://en.wikipedia.org/wiki/GNU_General_Public_License#Linking_and_derived_works) and [here](https://en.wikipedia.org/wiki/GNU_General_Public_License#Communicating_and_bundling_with_non-GPL_programs)), but I do understand that I don't have to worry about it until I publish source code. Wish I could help more!

Also, when I first submitted the Issue, I was simply looking for a `License.txt` file in the root directory of the project. I should have looked a bit more!

Once again, nice work!

**Comment by linqiaozhi** [comment]
After many months, finally got around to adding a LICENSE.txt, which compiles the licenses of all the software included in this repo. I also added MIT License for the new files I created.  It's in the fork (https://github.com/linqiaozhi/FIt-SNE/commit/e960b3c74e9d98f67409608aaaa5322366fcf1e2), will be merging soon.

Thanks for the help @aprive...and sorry it took a while to get to it! Let me know if you have any other thoughts.
